pyiron/vasp/base.py

```diff
@@ -400,6 +400,12 @@ def collect_output(self):
                 exclude_nodes=self._exclude_nodes_hdf,
             )
 
+    def nbands_convergence_check(self):
+        if (max(self["output/electronic_structure/occ_matrix"][:,-1])>0):
```

Comment:
I see only two indices but shouldn't there be three? I presume the first is k-points and the second is bands. Where's spin?

Reply:
And I guess we can make it shorter:

```
def nbands_convergence_check(self):
    return np.all(np.isclose(self["output/electronic_structure/occ_matrix"][:,-1], 0))
```

Reply:
Oh wait there should be even four indices since there should be also ionic steps.

Reply:
Thanks Sam for the comments, I ran a couple of calculations with spin and more than one ionic steps, the shape of the occupancy matrix does not change and remains the same as (kpoints, nbands).  I have made the function shorter as you suggested and added a docstring. 
If this satisfies your comments, you can resolve the conversation. Thanks.

Reply:
Ok I created a new issue and hopefully @sudarsan-surendralal will reply soon. Feel free to merge it now or after Sudarsan corrects the bug.

Reply:
The shape in `output/electronic_structure` should be `n_spin x n_kpoints x n_bands`. This is the electronic structure of the final step only. The spin not being there is a bug!

Reply:
I agree that the most interesting one is the final occupancy, but shouldn't we also check the occupancies of all steps? This being said, I admit it's possible that the amount of data to store might become too large if there are too many steps...